packages/core/entity/FileExplorerURL/index.ts

```diff
@@ -181,13 +185,15 @@ export default class FileExplorerURL {
         return `\`${filter.name}\`=="${filter.value}"`;
     }
 
-    private static convertCollectionToPython(collection: Collection | undefined) {
-        if (collection?.uri) {
-            const comment = "#Convert current datasource file to a pandas dataframe";
-            const extension = collection.uri.substring(collection.uri.lastIndexOf(".") + 1);
+    private static convertDataSourceToPython(source: Source | undefined) {
+        if (source) {
+            const comment =
+                "#Convert current datasource file to a pandas dataframe\n" +
+                "#You may need to manually update the path to the file";
+
             // Currently suggest setting all fields to strings; otherwise pandas assumes type conversions
             // TO DO: Address different non-string type conversions
-            const code = `df = pandas.read_${extension}('${collection.uri}').astype('str')`;
+            const code = `df = pd.read_${source.type}('${source.name}').astype('str')`;
```

Comment:
I think you want `source.uri` here

Reply:
There is only no uri when we query against AICS FMS

Reply:
Fixed in [9b14578](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/98/commits/9b14578d20d689b15227e9c03bd0adba2f3fbf5e)